crutest/src/main.rs

```diff
@@ -1470,17 +1491,62 @@ fn validate_vec<V: AsRef<[u8]>>(
                     res = ValidateStatus::Bad;
                 }
 
-                println!(
-                    "{}:{} bo:{} Volume offset:{}  Expected:{} Got:{}",
-                    msg,
-                    block_offset,
-                    i,
-                    byte_offset + i as u64,
-                    seed,
-                    data[data_offset + i],
-                );
+                // Check to see if we have an error already
+                block_err = match block_err {
+                    None => Some(BlockErr {
+                        block: block_offset,
+                        starting_offset: i,
+                        ending_offset: i,
+                        starting_volume_offset: byte_offset,
+                        ending_volume_offset: byte_offset,
+                        expected_value: seed,
+                        found_value: data[data_offset + i],
+                        status: res.clone(),
+                        msg,
+                    }),
+                    Some(mut be) => {
+                        // We have an existing error, does this extend our
+                        // range or start a new one?
+                        if be.status == res
+                            && be.found_value == data[data_offset + i]
+                            && be.ending_offset + 1 == i
+                        {
+                            be.ending_offset = i;
+                            be.ending_volume_offset = byte_offset;
+                            Some(be)
+                        } else {
+                            // Print the current range
+                            println!("{}", be);
+                            // Start a new range.
+                            Some(BlockErr {
+                                block: block_offset,
+                                starting_offset: i,
+                                ending_offset: i,
+                                starting_volume_offset: byte_offset,
+                                ending_volume_offset: byte_offset,
+                                expected_value: seed,
+                                found_value: data[data_offset + i],
+                                status: res.clone(),
+                                msg,
+                            })
+                        }
+                    }
+                };
```

Comment:
Nit: we can reduce duplication here by combining the `None` and non-matching cases
```suggestion
                block_err = match block_err {
                    // The error is contiguous with our current error
                    Some(mut be)
                        if be.status == res
                            && be.found_value == data[data_offset + i]
                            && be.ending_offset + 1 == i =>
                    {
                        be.ending_offset = i;
                        be.ending_volume_offset = byte_offset;
                        Some(be)
                    }
                    _ => {
                        // Print the current range, if it exists
                        if let Some(be) = block_err {
                            println!("{}", be);
                        }
                        // Start a new range.
                        Some(BlockErr {
                            block: block_offset,
                            starting_offset: i,
                            ending_offset: i,
                            starting_volume_offset: byte_offset,
                            ending_volume_offset: byte_offset,
                            expected_value: seed,
                            found_value: data[data_offset + i],
                            status: res.clone(),
                            msg,
                        })
                    }
                };
```